### PR TITLE
feat: install/uninstall modules from welcome window

### PR DESCRIPTION
**Pull request by erikas-taroza**
This PR will add functionality to the welcome window where users can install or uninstall Mirage modules. 
image
image

### CODE REVIEW

**Comment by uweenukr** [comment]
can you resolve the TODO code smell by updating the names? then it should be good to go

**Comment by erikas-taroza** [comment]
@uweenukr I will fix it when the modules have the namespace changed. Otherwise import will throw errors.

**Comment by erikas-taroza** [comment]
Some modules still need their package names updated. I could change them right now but things can break.

**Comment by paulpach** [comment]
Can you add a screenshot in the description?

**Comment by paulpach** [comment]
maybe, for now,  we should only display the modules that actually work.  Discovery is not ready,  Momentum is not ready,  websockets are not ready.   We can just add them once they are good to go.

Also, please make the scrollbar cover the entire right-hand side, and only show it when needed.

**Comment by paulpach** [comment]
Instead of "Modules",  let's change it to "Packages" to stay consistent with how unity calls them.

File: Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uxml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 <ui:UXML xmlns:ui="UnityEngine.UIElements" xmlns:uie="UnityEditor.UIElements" xsi="http://www.w3.org/2001/XMLSchema-instance" engine="UnityEngine.UIElements" editor="UnityEditor.UIElements" noNamespaceSchemaLocation="../../../../UIElementsSchema/UIElements.xsd" editor-extension-mode="False">
     <Style src="WelcomeWindow.uss" />
-    <ui:VisualElement name="Banner">
+    <ui:VisualElement name="Banner" style="background-color: rgb(11, 12, 11);">
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uxml:3]: split this into another PR,  this has nothing to do with installing packages

File: Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uxml [hunk 2]
```diff
@@ -17,7 +18,7 @@
         </ui:VisualElement>
         <ui:VisualElement name="Spacer" />
         <ui:VisualElement name="RightColumnBox">
-            <ui:VisualElement name="Welcome" style="flex-grow: 0; width: 100%; height: 100%;">
+            <ui:VisualElement name="Welcome" style="flex-grow: 0; width: 100%; height: 100%; display: none;">
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/Resources/WelcomeWindow.uxml:21]: was this commited by accident?

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 3]
```diff
@@ -30,15 +33,30 @@ public class WelcomeWindow : EditorWindow
         private const string BestPracticesUrl = "https://miragenet.github.io/Mirage/Articles/Guides/BestPractices.html";
         private const string FaqUrl = "https://miragenet.github.io/Mirage/Articles/Guides/FAQ.html";
         private const string SponsorUrl = "";
-        private const string DiscordInviteUrl = "https://discord.gg/rp6Fv3JjEz";
+        private const string DiscordInviteUrl = "https://discord.gg/DTBPBYvexy";
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:36]: separate PR,  unrelated to package installation

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 4]
```diff
@@ -30,15 +33,30 @@ public class WelcomeWindow : EditorWindow
...
+        //request for the module install
+        private static AddRequest installRequest;
+        private static RemoveRequest uninstallRequest;
+        private static ListRequest listRequest;
+
+        private static WelcomeWindow currentWindow;
+
         //window size of the welcome screen
         private static Vector2 windowSize = new Vector2(500, 415);
+        private static string screenToOpenKey = "MirageScreenToOpen";
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:56]: split into separate PR.  Unrelated to package install

File: Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs [hunk 5]
```diff
@@ -30,15 +33,30 @@ public class WelcomeWindow : EditorWindow
...
 
         //editorprefs keys
-        private static string firstStartUpKey = string.Empty;
+        private static string firstTimeVersionKey = string.Empty;
```
Comment (paulpach) [inline Assets/Mirage/Editor/WelcomeWindow/WelcomeWindow.cs:59]: split into separate PR

**Review by paulpach — CHANGES_REQUESTED** [review]
There are changes here that are unrelated to package installation.   Please move those changes to separate PRs.

**Pull request merged**